packages/next/export/worker.ts

```diff
@@ -215,8 +215,8 @@ export default async function exportPage({
         subFolders ? `${_path}${sep}index.html` : `${_path}.html`
       let htmlFilename = getHtmlFilename(filePath)
 
-      const pageExt = extname(page)
-      const pathExt = extname(path)
+      const pageExt = isDynamic ? '' : extname(page)
```

Comment:
```suggestion
      // dynamic routes can provide invalid extensions e.g. /blog/[...slug] returns an 
      // extension of `.slug]`
      const pageExt = isDynamic ? '' : extname(page)
```
Let's add a note here so we remember why we're doing this